src/geoUtils.ts

```diff
@@ -205,3 +211,101 @@ export function findByPostalCode(
+  switch (type) {
+    case "all":
+      return allTaxOffices[locale];
+    case "taxOffice":
+      return allTaxOffices[locale].map((taxOffice) => taxOffice.officialName);
+  }
+}
+
+type TaxOfficeOptionsById = {
+  id: number;
+  by?: "taxOfficeId" | "region" | "unit" | "municipality" | "postalCode";
+} & TaxOfficeOptions;
+
+export function getTaxOfficeBy(options: TaxOfficeOptionsById): TaxOffice | TaxOffice[] | undefined {
+  const { id, by = "taxOfficeId", locale = "el" } = options;
+  if (by === ("taxOfficeId" || "postalCode")) {
+    return allTaxOffices[locale].find((taxOffice) => {
+      if (by === "taxOfficeId" && id === taxOffice.id) {
+        return taxOffice;
+      }
+      if (taxOffice.postalCodes?.includes(id)) {
+        return taxOffice;
+      }
+      return;
+    });
+  } else {
+    return allTaxOffices[locale].filter((taxOffice) => {
+      if (by === "region" && taxOffice.relations.regionId === id) {
+        return taxOffice;
+      }
+      if (by === "unit" && taxOffice.relations.unitIds?.includes(id)) {
+        return taxOffice;
+      }
+      if (by === "municipality" && taxOffice.relations.municipalityIds?.includes(id)) {
+        return taxOffice;
+      }
+      return;
+    });
+  }
+}
+
+type TaxOfficeOptionsByTerm = { searchTerm?: string } & TaxOfficeOptions;
+
+export function searchTaxOffice(options: TaxOfficeOptionsByTerm = {}): TaxOffice[] | TaxOffice {
+  const { searchTerm, locale = "el" } = options;
+  if (!searchTerm) {
+    return allTaxOffices[locale];
+  }
+  // if ((locale === "el" && isGreekLatinMixed(searchTerm) !== "greek") || (locale === "en" && isGreekLatinMixed(searchTerm) !== "latin")) {
+  //   throw new Error('Search term and localization missmatch');
+  // }
+  const normalizedTerm = normalizeString(searchTerm);
+  return allTaxOffices[locale].filter((taxOffice) => {
+    return normalizeString(taxOffice.name).includes(normalizedTerm) ? taxOffice : "";
+  });
+}
+
+//Removes accents,spaces and special chars from the string
+export function normalizeString(input: string): string {
+  const accentChars = /[άέήίόύώΆΈΉΊΌΎΏϊΐΪϋΰΫ]/g;
+  const accentReplacements: { [key: string]: string } = {
+    ά: "α",
+    έ: "ε",
+    ή: "η",
+    ί: "ι",
+    ό: "ο",
+    ύ: "υ",
+    ώ: "ω",
+    Ά: "Α",
+    Έ: "Ε",
+    Ή: "Η",
+    Ί: "Ι",
+    Ό: "Ο",
+    Ύ: "Υ",
+    Ώ: "Ω",
+    ϊ: "ι",
+    ΐ: "ι",
+    Ϊ: "Ι",
+    ϋ: "υ",
+    ΰ: "υ",
+    Ϋ: "Υ",
+  };
+  let normalized = input.replace(accentChars, (match) => accentReplacements[match] || match);
+  normalized = normalized.replace(/[ \-_!@#$%^&*()]/g, ""); // Remove spaces and special characters
+  return normalized.toLocaleUpperCase(); // Convert to uppercase
+}
+
+//Compare greek words
+export function compareGreekStrings(stringA: string, stringB: string): boolean {
+  if (normalizeString(stringA) === normalizeString(stringB)) {
+    return true;
+  }
+  return false;
```

Comment:
You can simply return  replace thebelow 4 lines with:

```suggestion
  return normalizeString(stringA) === normalizeString(stringB));
```